### PR TITLE
Send update metadata about update plans

### PR DESCRIPTION
**Pull request by Frassle**
# Description

Send update metadata to the service about if we're using update plans.

## Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works

- [ ] I have run `make changelog` and committed the `changelog/pending/<file>` documenting my change

- [ ] Yes, there are changes in this PR that warrants bumping the Pulumi Service API version

### CODE REVIEW

File: pkg/cmd/pulumi/preview.go [hunk 1]
```diff
@@ -138,7 +138,7 @@ func newPreviewCmd() *cobra.Command {
 				return result.FromError(err)
 			}
 
-			m, err := getUpdateMetadata(message, root, execKind, execAgent)
+			m, err := getUpdateMetadata(message, root, execKind, execAgent, "")
```
Comment (Frassle) [inline pkg/cmd/pulumi/preview.go:141]: Should we send something here if we're using preview to generate a plan? "update_plan": "generating"?

**Comment by EvanBoyle** [comment]
For context, here is the page these values are displayed in the console: 

image

**Review by EvanBoyle — COMMENTED** [review]
I don't have full context here, but I think a reasonable place to start would be setting `updatePlan: true` if an only if an update plan is provided. 

We can iterate on this in the future as well.

File: pkg/cmd/pulumi/preview.go (hunk 1 above)
Comment (EvanBoyle) [reply pkg/cmd/pulumi/preview.go:141]: I think we could use the same field `updatePlane: true` if we are generating an update plan as a part of the preview.

File: pkg/cmd/pulumi/up.go [hunk 2]
```diff
@@ -96,7 +97,15 @@ func newUpCmd() *cobra.Command {
 			return result.FromError(err)
 		}
 
-		m, err := getUpdateMetadata(message, root, execKind, execAgent)
+		var updatePlan string
+		if hasExperimentalCommands() {
+			updatePlan = constant.UpdatePlanExperimental
```
[the remark this answers is not in the document]
Comment (Frassle) [reply pkg/cmd/pulumi/up.go:102]: It is, on the basis that if your in experimental mode we'd use the plan generated by the "up" preview stage to constrain the actual update stage. If they _also_ passed --plan that would take priority and we'd send "explicit" here.

I think this three state makes sense today, but long term plans it probably does make more sense to just make this a true/false.

**Review by AaronFriel — APPROVED** [review]
bors merge

**Pull request merged**